src/deleted.psm1

```diff
@@ -0,0 +1,975 @@
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+Remove-SafeguardDeletedAsset -AccessToken $token -Appliance 10.5.32.54 -Insecure 5
+
+.EXAMPLE
+Remove-SafeguardDeletedAsset 5
+#>
+function Remove-SafeguardDeletedAsset
+{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$AssetToDelete
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    $local:AssetId = (Resolve-SafeguardDeletedAssetId -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure $AssetToDelete)
+    Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core DELETE "Deleted/Assets/$($local:AssetId)"
+}
+
+<#
+.SYNOPSIS
+Restore a soft-deleted asset to its previous (un-deleted) state.
+
+.DESCRIPTION
+Restore a soft-deleted asset to its previous (un-deleted) state. Restoring an asset does not
+automatically restore soft-deleted asset accounts associated with the asset.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER AssetToRestore
+An object representing the deleted asset to restore or an integer 
+containing the ID of the asset account or a string containing the name. When
+AssetToRestore is an object returned by Get-SafeguardDeletedAsset, any modifications 
+to property values will take precedent when restoring the asset.
+
+.INPUTS
+None.
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+Restore-SafeguardDeletedAsset -AccessToken $token -Appliance 10.5.32.54 -Insecure 5
+
+.EXAMPLE
+Restore-SafeguardDeletedAsset 5
+#>
+function Restore-SafeguardDeletedAsset
+{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$AssetToRestore
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    $local:Asset = $AssetToRestore
+    if($AssetToRestore -is [int] -or $AssetToRestore -is [string]) {
+        $local:Asset = (Get-SafeguardDeletedAsset -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure $AssetToRestore)[0]
```

Comment:
Do we need `[0]` here? $local:Asset would be a single object and not an array of objects because $AssetToGet parameter is being passed in the call to  Get-SafeguardDeletedAsset

Reply:
Get-SafeguardDeletedAsset returns an array